### ISSUE TITLE
Investigate error with activity file XML import to IATI Publisher

### ISSUE THREAD

**Issue by emmajclegg** [younginnovations/iatipublisher#1463]
This affects a single publisher - the details can be found in this Zendesk support ticket: https://iati.zendesk.com/agent/tickets/44547

**Comment by sarinasindurakar** [comment]
Hi @emmajclegg ,
The Initial issue while importing csv file listed in (https://github.com/younginnovations/iatipublisher/issues/1462) has been resolved and deployed to staging .

We have tested the CSV file and found out that it contains invalid data. The  activity elements : participating organization type, CRS channel, organization role, recipient country, activity date, and status all appear to have incorrect data.
![Image](https://github.com/younginnovations/iatipublisher/assets/65848732/39ef37dc-dc0d-45c0-8ece-f35a79f82cbb)

**Comment by PG-Momik** [comment]
@emmajclegg A quick update on this, 

We've looked into the the issues mentioned in the ticket, namely:
1. Files not being imported
2. Activity duplication

### Files not being imported
Recent changes made in #1407  was missing a null  check when resolving if any elements were using deprecated code. I've added the necessary null check. This issue has been fixed and is being tested. 

### Activity duplication
We've confirmed that some activities have indeed been duplicated. All of the duplicated activities were uploaded to the system via xml. We're looking into recreating the duplication issue. We haven't been able to pin point the exact cause for duplication itself. I'll keep you updated on the progress.


cc: @praweshsth @sarinasindurakar

**Comment by PG-Momik** [comment]
Hi @emmajclegg. Futher updates on this.

### Files not being imported
The null check required to fix this issue has been deployed to production alongside #1462 . The issue mentioned in the Zendesk ticket will not occur not. Please let us know if the issue persists.

### Activity duplication
We've found the cause of activity duplication when being updated via XML. The code to check if an activity being imported via XML already exists in database is failing.
If a publisher tries to update/import an already existing activity via XML, but has added extra spaces in the `<iati-identifier>` field.
 EXAMPLE: `<iati-identifier>IDENTIFER{extra space}</iati-identifier>`
The system considers this a new activity, resulting in the duplication.

We'll look into if any more publishers have similar issues and let you know of the possible solutions.


---
cc: @praweshsth @sarinasindurakar

**Comment by emmajclegg** [comment]
Thanks for the update @PG-Momik .

### File not being imported
Thanks for applying the fix - I've tested the XML file upload on my test account and it looks to be working. I've let the publisher know.

### Activity duplication
When you can, please send me a list of the duplicate activities (either by email or in this support ticket: https://iati.zendesk.com/agent/tickets/44548) and I'll work with the publisher to remove the duplicates. If there is an automated way that you can see if activities are the same apart from spaces in their identifier(?), please let me know as that would make a bulk deletion easier your side.

**Comment by sarinasindurakar** [comment]
@PG-Momik 

- [ ] 1. When the user imports the file from an XML file and adds spaces before and after the identifier, the file is imported, but a random number is being set in the IATI identifier.

![Image](https://github.com/younginnovations/iatipublisher/assets/65848732/d51357fa-7ba0-4c90-9274-086a0f63a955)

- [ ] 2. Activities are still being duplicated when the user imports the file with added spaces.

**Comment by PG-Momik** [comment]
@emmajclegg  this has been deployed to staging.

**Comment by PG-Momik** [comment]
Hi @emmajclegg , 
I've reopened this issue to address the concerns raised in the latest reply in [Zendesk ticket 44548](https://iati.zendesk.com/agent/tickets/44548).

> but only 876 activities in the downloaded activity list from IATI Publisher (also attached). Are there more duplicate activities in IATI Publisher? 

Yes there seems to be more duplicate activities in IATI publisher than initially reported. The database instance that I initially used to troubleshoot this issue did not contain the activities that have been repeated now.  Looking at the created_at timestamp, it seems these duplicate activities were created, after the list had been mailed to you and before the fix had been deployed to production. I've mailed you a new list of duplicate activities.
